src/lib.rs

```diff
@@ -123,21 +175,40 @@ impl Entry {
     }
 }
 
-pub struct Structures<'a, 'b> {
-    entry: &'a Entry,
+/// An iterator that traverses the SMBIOS structure tables.
+/// This struct is produced by the `structures` method on `Entry`. See its documentation for more details.
+#[derive(Clone, Debug, Eq, Hash, PartialEq)]
+pub struct Structures<'entry, 'buffer> {
+    entry: &'entry Entry,
```

Comment:
Maybe it would be better to clone the entry and avoid the lifetime complexity, it's pretty small anyway

Reply:
The entry can actually be copied (yay, packed data! ... not really yay, that is pretty terrible). I might change it around then and not store it inside the RawStructure, because that would get pretty expensive, but instead pass it in by reference to the constructors (and it is up to the user who is decoding the unknown ones to decode with the EntryPoint if they need it)... 

Or, maybe it is better to pass on the version number to the structures for parsing. Thoughts?